monai/transforms/composables.py

```diff
@@ -120,6 +121,61 @@ def __call__(self, data):
         return d
 
 
+@export
+class RandCropByPosNegLabeld(Randomizable, MapTransform):
+    """
+    Crop random fixed sized regions with the center being a foreground or background voxel
+    based on the Pos Neg Ratio.
+    And will return a list of dictionaries for all the cropped images.
+
+    Args:
+        keys (list): parameter will be used to get and set the actual data item to transform.
+        label_key (str): name of key for label image, this will be used for finding foreground/background.
+        size (list, tuple): the size of the crop region e.g. [224,224,128]
+        pos (int, float): used to calculate the ratio ``pos / (pos + neg)`` for the probability to pick a
+          foreground voxel as a center rather than a background voxel.
+        neg (int, float): used to calculate the ratio ``pos / (pos + neg)`` for the probability to pick a
+          foreground voxel as a center rather than a background voxel.
+        batch_size (int): number of samples (crop regions) to take in each batch.
+    """
+
+    def __init__(self, keys, label_key, size, pos=1, neg=1, batch_size=1):
+        MapTransform.__init__(self, keys)
+        assert isinstance(label_key, str), 'label_key must be a string.'
+        assert isinstance(size, (list, tuple)), 'size must be list or tuple.'
+        assert all(isinstance(x, int) and x > 0 for x in size), 'all elements of size must be positive integers.'
+        assert float(pos) >= 0 and float(neg) >= 0, "pos and neg must be greater than or equal to 0."
+        assert float(pos) + float(neg) > 0, "pos and neg cannot both be 0."
+        assert isinstance(batch_size, int), "invalid batch size: {}. batch_size must be an integer.".format(batch_size)
+        assert batch_size >= 0, 'batch_size must be greater than or equal to 0.'
+        self.label_key = label_key
+        self.size = size
+        self.pos_ratio = float(pos) / (float(pos) + float(neg))
+        self.batch_size = batch_size
+
+    def __call__(self, data):
+        d = dict(data)
+        label = d[self.label_key]
+        centers = generate_pos_neg_label_crop_centers(label, self.size, self.batch_size, self.pos_ratio, self.R)
```

Comment:
would be great to keep a record of the `centers` after generating the patches, i think we could make `centers` an instance variable, or make new keys in the output data

Reply:
Hi @wyli ,

Thanks for your comments, I already updated code to save centers and use randomize API.